src/main/java/org/opensearch/knn/index/memory/NativeMemoryEntryContext.java

```
@@ -278,6 +363,11 @@ public Integer calculateSizeInKB() {
             return size;
         }
 
+        @Override
+        public void openVectorIndex() {
+            return;
+        }
```

this function should also throw the `UnsupportedOperationException` just like the other class.

---

Since we are offloading the opening of vector index from the `load` method in loadStrategy, this call ends up calling the `openVectorIndex`  of the corresponding entryContext here - https://github.com/Gankris96/k-NN/blob/ecdb8fa6c23992bac07170cc54e347ed3f2bc85f/src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java#L357 

This means that we can't throw `UnsupportedOperationException` here. I can either leave the return here as a no-op for the non vector cases or i have to scope the openVectorIndex call in the above line i shared to only do so if `nativeMemoryEntryContext instanceof NativeMemoryEntryContext.IndexEntryContext`. This is needed because if `forceEviction` is not enabled, then we directly jump to the load call, but the graph file is not yet opened.